File: DotNetMessaging/SpringBootMessaging/src/main/java/com/messaging/api/models/Message.java
package com.messaging.api.models;

import lombok.Data;
import lombok.EqualsAndHashCode;
import org.springframework.data.mongodb.core.mapping.Document;

@Data
@EqualsAndHashCode(callSuper = true)
@Document(collection = "messages")
public class Message extends BaseEntity {
    private String chatId;
    
    private String groupId;
    
    private String senderId;
    
    private String content;
    
    private MessageType type = MessageType.Text;
    
    private String mediaUrl;
    
    private String mediaType;
    
    private String mediaFileName;
    
    private Long mediaSize;
    
    private String replyToMessageId;
    
    private boolean isDeleted = false;
}

File: DotNetMessaging/backend/DotNetMessaging.API/Models/Message.cs
using MongoDB.Bson.Serialization.Attributes;

namespace DotNetMessaging.API.Models;

public class Message : BaseEntity
{
    [BsonElement("chatId")]
    public string? ChatId { get; set; }

    [BsonElement("groupId")]
    public string? GroupId { get; set; }

    [BsonElement("senderId")]
    public string SenderId { get; set; } = string.Empty;

    [BsonElement("content")]
    public string Content { get; set; } = string.Empty;

    [BsonElement("type")]
    [BsonRepresentation(MongoDB.Bson.BsonType.String)]
    public MessageType Type { get; set; } = MessageType.Text;

    [BsonElement("mediaUrl")]
    public string? MediaUrl { get; set; }

    [BsonElement("mediaType")]
    public string? MediaType { get; set; }

    [BsonElement("mediaFileName")]
    public string? MediaFileName { get; set; }

    [BsonElement("mediaSize")]
    public long? MediaSize { get; set; }

    [BsonElement("replyToMessageId")]
    public string? ReplyToMessageId { get; set; }

    [BsonElement("isDeleted")]
    public bool IsDeleted { get; set; } = false;
}

public enum MessageType
{
    Text,
    Image,
    Video,
    Audio,
    Document,
    Location
}
